packages/core/src/account/types.ts

```diff
@@ -136,4 +136,28 @@ export interface ISmartContractAccount<
    * @returns the address of the entry point contract for the smart account
    */
   getEntryPointAddress(): Address;
+
+  /**
+   * Allows you to add additional functionality and utility methods to this account
+   * via a decorator pattern.
+   *
+   * NOTE: this method does not allow you to override existing methods on the provider.
+   *
+   * @example
+   * ```ts
+   * const account = new BaseSmartCobntractAccount(...).extend((account) => ({
+   *  debugSendUserOperation: (...args) => {
+   *    console.log("debugging send useroperation");
+   *    return this.rpcProvider.sendUserOperation(...args);
+   *  }
```

Comment:
maybe we should make this like:

```suggestion
   *  readAccountState: async (...args) => {
   *    return this.rpcProvider.readContract({
   *        address: await this.getAddress(),
   *        abi: ThisContractsAbi
   *        args: args
   *    });
   *  }
```